simplq/src/components/pages/Join/index.jsx

```diff
@@ -35,15 +31,17 @@ export default ({ history, match }) => {
     <div>
       <HeaderSection queueName={queueStatus.queueName} history={history} />
       <div className={styles['main-content']}>
-        <div className={styles['queue-stats']}>
-          <QueueStats queueStatus={queueStatus} />
-        </div>
-        <p className={styles['message']}>Please enter your contact details to join this queue</p>
-        <JoinQueueForm
-          queueId={queueId}
-          joinQueueHandler={joinQueueHandler}
-          buttonText="Join Queue"
-        />
+        <LoadingStatus dependsOn="getQueueStatusByName">
```

Comment:
Should we also define `getQueueStatusByName` etc as constants somewhere and use it?

Reply:
I didn't like declaring, exporting, importing another variable. Just pass the whole action in an object.

```js
<LoadingStatus dependsOn={{ getQueueStatusByName }}>
```

Reads nicely: **Loading status depends on qetQueueStatusByName** 😄